### PR TITLE
Avoid using ivars in helpers too

### PR DESCRIPTION
**Pull request by bockets**

### CODE REVIEW

File: _patterns/no-instance-vars-in-partials-or-helpers.md [hunk 1]
```diff
@@ -61,3 +58,29 @@ Since all instance vars have `nil` value assigned by default, the exception migh
...
+```ruby
+def show_free_ebook_ad?(forum:)
+  forum_ids.include?(forum.category_id)
+end
+```
+
+```ruby
+def show_free_ebook_ad?
+  forum_ids.include?(current_forum.category_id)
+end
```
Comment (sebsogamoso) [inline _patterns/no-instance-vars-in-partials-or-helpers.md:83]: What do you think about not using a "magic" `forum_ids` method in these examples? Even if a bit simpler, I think not having that in the example better illustrates the point.
```suggestion
def show_free_ebook_ad?
  @forum.category == "books"
end
```

What happens if the `@forum` instance variable is renamed? It's also unclear what type of object the `@forum` instance variable refers to.

## Good

```ruby
def show_free_ebook_ad?(forum:)
  forum.category == "books"
end
```

```ruby
def show_free_ebook_ad?
  current_forum.category == "books"
end
```

**Review by sebsogamoso — APPROVED** [review]
I like this but I think this should be a separate rule from _ivars_ in partials. They are similar but not the same, keeping each rule small and simple makes them a bit easier to understand.

File: _patterns/no-instance-vars-in-partials-or-helpers.md (hunk 1 above)
Comment (bockets) [reply _patterns/no-instance-vars-in-partials-or-helpers.md:83]: Ah yea! Good point thanks 🙏

**Comment by SpiderMasterMind** [comment]
I like this - or in the way @sebsogamoso suggests as well, I personally don't mind

**Review by cernivec — APPROVED** [review]

**Pull request merged**